ob1k-core/src/main/java/com/outbrain/ob1k/client/ClientBuilder.java

```
@@ -208,8 +206,7 @@ private static RequestMarshallerRegistry createRegistry(final Class type) {
           final List<StreamFilter> filters = mergeFilters(StreamFilter.class, streamFilters, endpointDescriptor.filters);
           clientEndpoint = new StreamClientEndpoint(httpClient, registry, endpoint, filters.toArray(new StreamFilter[filters.size()]));
         } else {
-          final List<SyncFilter> filters = mergeFilters(SyncFilter.class, syncFilters, endpointDescriptor.filters);
-          clientEndpoint = new SyncClientEndpoint(httpClient, registry, endpoint, filters.toArray(new SyncFilter[filters.size()]));
+          throw new IllegalArgumentException("Endpoint methods return type must be either ComposableFuture or Observable.");
```

Do we really want to throw an exception here?